src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1453,8 +1460,8 @@ static UCS_CLASS_INIT_FUNC(uct_dc_mlx5_iface_t, uct_md_h tl_md, uct_worker_h wor
 
     uct_dc_mlx5_iface_init_version(self, tl_md);
 
-    self->tx.ndci                          = config->ndci;
-    self->tx.policy                        = (uct_dc_tx_policy_t)config->tx_policy;
+    self->tx.ndci                          = num_dcis;
+    self->tx.policy                        = tx_policy;
```

Comment:
maybe remove num_dcis, tx_policy vars and use self->tx.ndci, self->tx.policy instead?